lib/IRGen/ClassMetadataVisitor.h

```diff
@@ -185,6 +192,21 @@ template <class Impl> class ClassMetadataVisitor
     asImpl().addVTableEntries(theClass);
   }
 
+  /// Add fields associated with the given class and its bases.
+  void addEmbeddedClassMembers(ClassDecl *theClass) {
+    // Visit the superclass first.
+    if (auto *superclassDecl = theClass->getSuperclassDecl()) {
+      addEmbeddedClassMembers(superclassDecl);
+    }
+
+    // Note that we have to emit a global variable storing the metadata
+    // start offset, or access remaining fields relative to one.
+    asImpl().noteStartOfImmediateMembers(theClass);
+
+    // Add vtable entries.
+    asImpl().addVTableEntries(theClass);
```

Comment:
Do we want the full vtable?

Reply:
Can we be more lazy and just emit the methods that are used? Or are we assuming that lto will strip them?

Reply:
I'm actually assuming that SILOptimizer (namely PruneVTables and DeadFunctionElimination) will take care of that, because we can internalize public declarations in embedded Swift.